sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/output.py

```diff
@@ -73,9 +73,10 @@ def __init__(self, *, type=AssetTypes.URI_FOLDER, path=None, mode=None, descript
         # As an annotation, it is not allowed to initialize the name.
         # The name will be updated by the annotated variable name.
         self.name = None
+        self.asset_name = kwargs.pop('asset_name', None)
         self._is_primitive_type = self.type in IOConstants.PRIMITIVE_STR_2_TYPE
         self.description = description
-
+        self.asset_version = kwargs.pop('asset_version', None)
```

Comment:
As this output class is also works for standalone job, does other standalone job type works? May give command job a try.

Reply:
CommandJob support register output. Add test `test_command_job_register_output`.